homeassistant/components/cast/media_player.py

```diff
@@ -502,6 +511,60 @@ def media_seek(self, position):
         media_controller = self._media_controller()
         media_controller.seek(position)
 
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Implement the websocket media browsing helper."""
+        media: Media = await self.hass.components.media_source.async_find_media(
+            media_content_id
+        )
+        if not media:
+            raise BrowseError("Could not find media for given path.")
+
+        def media_to_dict(media: Media):
+            """Convert Media class to browse media dictionary."""
+            response = {
+                "title": media.name,
+                "media_content_type": media.mime_type or "",
+                "media_content_id": media.location,
+                "can_play": media.is_file,
+                "can_expand": media.is_dir,
+            }
+
+            if media.children:
+                response["children"] = [
+                    media_to_dict(child) for child in media.children
+                ]
+
+            return response
+
+        return media_to_dict(media)
+
+    async def async_play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        # Handle media_source
+        if media_id.startswith(URI_SCHEME):
+            matches = re.match(URI_SCHEME_REGEX, media_id)
+            media_id = matches.group("path")
+
+            # Sign URL with Home Assistant Cast User
+            config_entries = self.hass.config_entries.async_entries(CAST_DOMAIN)
+            if config_entries:
+                user_id = config_entries[0].data["user_id"]
+                user = await self.hass.auth.async_get_user(user_id)
+                if user.refresh_tokens:
+                    refresh_token: RefreshToken = list(user.refresh_tokens.values())[0]
+
+                    media_id = async_sign_path(
+                        self.hass, refresh_token.id, media_id, timedelta(minutes=5),
+                    )
+
+            # prepend external URL
+            hass_url = get_url(self.hass, prefer_external=True)
+            media_id = f"{hass_url}{media_id}"
+
+        await self.hass.async_add_job(
```

Comment:
```suggestion
        await self.hass.async_add_executor_job(
```

Reply:
This wasn't addressed.

Reply:
Sorry, I thought Paulus addressed it in his followup commits. Either way, this is exactly what the parent does.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/media_player/__init__.py#L600-L604

I can add a follow-up PR to change it there and simply call `super` in this class.

Reply:
Yes, please. 👍